web/src/labs/marked/index.tsx

```diff
@@ -1,67 +1,75 @@
 import { matcher } from "./matcher";
 import { blockElementParserList, inlineElementParserList } from "./parser";
 
-export const marked = (
-  markdownStr: string,
-  blockParsers = blockElementParserList,
-  inlineParsers = inlineElementParserList
-): string | JSX.Element => {
-  for (const parser of blockParsers) {
-    const matchResult = matcher(markdownStr, parser.regexp);
-    if (!matchResult) {
-      continue;
-    }
-    const matchedStr = matchResult[0];
-    const retainContent = markdownStr.slice(matchedStr.length);
-
-    if (parser.name === "br") {
-      return (
-        <>
-          {parser.renderer(matchedStr)}
-          {marked(retainContent, blockParsers, inlineParsers)}
-        </>
-      );
-    } else {
-      if (retainContent === "") {
-        return parser.renderer(matchedStr);
-      } else if (retainContent.startsWith("\n")) {
-        return (
-          <>
-            {parser.renderer(matchedStr)}
-            {marked(retainContent.slice(1), blockParsers, inlineParsers)}
-          </>
-        );
-      }
-    }
-  }
+type Parser = typeof blockElementParserList[0];
```

Comment:
We can declare its structure instead of using `typeof blockElementParserList[0]`